### PR TITLE
Fix #3628: Rename sphinx_themes entry point to sphinx.html_themes

### LINKED ISSUE #3628: Rename sphinx_themes entry point to sphinx.themes
As discussed in #2803, `sphinx_themes` entry point should be renamed to `sphinx.themes`.

### PR DESCRIPTION
**Pull request by tk0miya**
refs: #3628 

* [x] Update docs
* [x] Do testing (on local?)

### CODE REVIEW

**Comment by tk0miya** [comment]
I'd tested this on my local. I feel it is hard to test the entry_points on our test script. So I don't add the test about it.
Thanks,

File: doc/theming.rst [hunk 1]
```diff
@@ -310,6 +290,44 @@ Python :mod:`ConfigParser` module) and has the following structure:
   and are accessible from all templates as ``theme_<name>``.
 
 
+.. _distribute-your-theme:
+
+Distribute your theme as a python package
+-----------------------------------------
```
Comment (tk0miya) [inline doc/theming.rst:296]: I think it would be nice if we provide a template for themes (or quickstart for themes!).

File: doc/theming.rst [hunk 2]
```diff
@@ -310,6 +290,44 @@ Python :mod:`ConfigParser` module) and has the following structure:
...
+            ]
+        },
+        ...
+    )
+
+    # 'your_package.py'
+    from os import path
+
+    def setup(app):
+        app.add_html_theme('name_of_theme', path.abspath(path.dirname(__file__)))
```
Comment (shimizukawa) [inline doc/theming.rst:320]: Is name needed?
If theme extension have multiple theme directory in single parent `theme_path`, the extension will call `app.add_html_theme(...)` for each theme **name** with one **theme_path**. Is it OK as a usage?

Additionally, in such case, does user need specify `name_of_theme` for each themes for `app.add_html_theme` and `sphinx.html_themes` ?

I think multiple themes in single package is not common case, however it is better to cover such case in this document.

**Review by shimizukawa — COMMENTED** [review]
commented.

File: doc/theming.rst (hunk 2 above)
Comment (tk0miya) [reply doc/theming.rst:320]: From theme-extension's sight, the name parameter is not needed. The name is already defined in entry point. So this parameter is duplicated. In addition, I think almost theme package might be contains only one theme. A package having multiple themes might be rare.

On the other hand, from API's sight, the name is important. Our theme manager looks up a theme using its name. So it is hard to look up if we drop the parameter.

Comment (tk0miya) [reply doc/theming.rst:320]: As a result of discussion in hack-day, I updated doc to follow about the case single package contains multiple themes.
@shimizukawa could you check this please?

**Review by shimizukawa — APPROVED** [review]
LGTM!

**Pull request merged**